lib/queue.js

```diff
@@ -1,27 +1,59 @@
 class Queue {
+    #front;
+    #rear;
+
     constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+        this.store = new Array(20);
+        // front and rear are index refs
+        this.#front = this.#rear = 0;
     }
 
-    enqueue(element) {
-        throw new Error("This method has not been implemented!");
+    enqueue(element) {        
+        if (
+            (this.#front === 0 && this.#rear === this.size() - 1)
+        ) {
+            throw 'QueueFullException'
+        }
+        else if (this.isEmpty()) {
+            this.store[this.#front] = element;
+        } else {
+            this.#rear += 1;
```

Comment:
👀 Make sure you're handling the case where `rear` wraps around!
```suggestion
            this.#rear = (this.#rear + 1) % this.capacity;
```